node/builder.go

```diff
@@ -299,7 +299,7 @@ func (nc *Builder) build(ctx context.Context) (*Node, error) {
 	nd.PorcelainAPI = porcelain.New(plumbing.New(&plumbing.APIDeps{
 		Bitswap:       bswap,
 		Chain:         chainState,
-		Sync:          cst.NewChainSyncProvider(chainSyncer),
+		Sync:          cst.NewChainSyncProvider(chainSyncer, bs, peerHost.ID()),
```

Comment:
The addition of the blockstore will likely be the most controversial change in this PR, here's why we are doing it: https://github.com/filecoin-project/go-filecoin/pull/3473/files#r327391073